amr-wind/physics/TaylorGreenVortex.cpp

```diff
@@ -31,17 +31,23 @@ void TaylorGreenVortex::initialize_fields(
 
         const auto& dx = geom.CellSizeArray();
         const auto& problo = geom.ProbLoArray();
+        const auto& probhi = geom.ProbHiArray();
+
         auto vel = velocity.array(mfi);
 
         amrex::ParallelFor(
             vbx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
                 const amrex::Real x = problo[0] + (i + 0.5) * dx[0];
                 const amrex::Real y = problo[1] + (j + 0.5) * dx[1];
                 const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
-                vel(i, j, k, 0) = std::sin(two_pi() * x) *
-                                  std::cos(two_pi() * y) * cos(two_pi() * z);
-                vel(i, j, k, 1) = -std::cos(two_pi() * x) *
-                                  std::sin(two_pi() * y) * cos(two_pi() * z);
+                const amrex::Real Lx = probhi[0]-problo[0];
+                const amrex::Real Ly = probhi[1]-problo[1];
+                const amrex::Real Lz = probhi[2]-problo[2];
```

Comment:
Move this out of the innermost loop to outside the `MFIter`. You can query the domain from mesh and pre-compute this.